pkg/handlers/router.go

```diff
@@ -73,6 +73,11 @@ func (r *Router) WithSearchSymbolsHandler(handler http.Handler) *Router {
 	return r
 }
 
+func (r *Router) WithDocumentOutlineHandler(handler http.Handler) *Router {
+	r.Handle("/projects/{id}/outline/{path:.*}", handler).Methods(http.MethodGet)
```

Comment:
according to restful design, the path should be something like `/projects/{id}/files/{path}/outline` because outline is a sub-resource of a file

Reply:
That was my initial take but `{path}/outline` does not work `path/something/outline` is a valid path, basically everything after `file/` will be interested as path.

Reply:
Does the order in registering handlers matter? can we register the outline handler first?

Reply:
Worst case we can use negative lookahead in files handlers to ensure the path doesn't end with `/symbols`

Reply:
Maybe a flag on `r.Handle("/projects/{id}/outline/{path:.*}", handler).Methods(http.MethodGet)`